src/metatensor/models/soap_bpnn/model.py

```diff
@@ -292,7 +298,7 @@ def forward(
                     "the model's capabilities."
                 )
 
-        soap_features = self.soap_calculator(systems)
+        soap_features = self.soap_calculator(systems, selected_samples=selected_atoms)
```

Comment:
For now, the naming is a bit different between rascaline and metatensor atomistic systems, so you'll need to do something like this:

https://github.com/Luthaf/rascaline/blob/master/python/rascaline-torch/tests/export.py#L51-L54